File: cluster_3bc/src/python/producer.py
from kafka import KafkaProducer

producer = KafkaProducer(bootstrap_servers='localhost:9092')

# send(<topic>, <message>)
# message must be bytes array
key = 0
while msg := input('Message: '):
    producer.send(
        topic='test2', 
        value=bytes(msg, 'utf-8'),
        key=bytes(str(key), 'utf-8')
    )
    key+=1

File: cluster_3bc/src/nodejs/producer.js
const {Kafka} = require('kafkajs');
const prompt = require('prompt-sync')();

const TOPIC_NAME = 'test';

const kafka = new Kafka({
    clientId: 'demo-producer',
    brokers: ['localhost:9092']
});

const producer = kafka.producer();

const run = async () => {
    await producer.connect();
    let message = "";
    do {
        message = prompt('Votre message (vide pour stop)?');
        if (message !== '') {
            await producer.send({
                topic: TOPIC_NAME,
                messages: [
                    {value: message},
                ],
            })
        }
    }
    while (message);
    await producer.disconnect()
}

run().catch(console.error)